pkg/tiller/release_server.go

```diff
@@ -59,6 +59,8 @@ var (
 	errMissingRelease = errors.New("no release provided")
 	// errInvalidRevision indicates that an invalid release revision number was provided.
 	errInvalidRevision = errors.New("invalid release revision")
+	//errInvalidName indicates that an invalid release name was provided
+	errInvalidName = errors.New("invalid release name")
```

Comment:
@bacongobbler  Many thanks for you review.  It a good idea, but   I worried is that it is return too long of the message.

Such as return  this error message:

errInvalidName = errors.New("invalid release name; must match  regex  ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$\  and the max length is 53 ")